### DISCUSSION TITLE
Is it possible to select a random song from an imported playlist each time the alarm plays? (for repeating alarms)

### DISCUSSION THREAD

**Discussion by TechPro424** [turtton/YtAlarm#195] [Q&A] [answered]
Is it possible to select a random song from an imported playlist each time the alarm plays? (for repeating alarms)

**Comment by turtton** [comment]
Thank you for opening discussions.
Currently, playlists is shuffled only one time, but I am planned to change this behavior. (However, I do not have enough time to work for this project for now😭

Related code:
https://github.com/turtton/YtAlarm/blob/52bad1a16de595fa54a2fa40699c74634e2e77f3/app/src/main/kotlin/net/turtton/ytalarm/ui/fragment/FragmentVideoPlayer.kt#L238-L239

**Reply by TechPro424** [reply]
When does this happen?

**Reply by turtton** [reply]
At alarm activation.

**Reply by TechPro424** [reply]
So each time the alarm rings?

**Reply by turtton** [reply, marked as answer]
Yes. Playlist is randomized before alarm start if you enable Shuffle option, and never changed order until you stop.

**Reply by TechPro424** [reply]
Great, downloading the app now!

**Reply by turtton** [reply]
I am sorry to say that this app currently does not work properly because its internal system is not compatible with the latest Youtube. I will update it soon. Please be patient.

**Reply by TechPro424** [reply]
I could try to fork and update it; I have experience with Java, and Kotlin is pretty similar to it. Which part is not working, exactly?

**Reply by turtton** [reply]
This app using yt-dlp so it needed to upgrade to latest version.
I just did that, but could not confirm it works on emulator, so could you try it too?.(Requires Android Studio beta)

**Reply by TechPro424** [reply]
What exactly does not work? Is there an error?

**Reply by turtton** [reply]
I opened issue -> https://github.com/turtton/YtAlarm/issues/196
